### PR TITLE
bluej: 5.0.3 -> 5.1.0

### PR DESCRIPTION
**Pull request by chvp**
###### Description of changes

https://bluej.org/versions.html

Closes #192713

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chvp** [comment]
Note that this package is currently broken in nixpkgs trunk because javafx is disabled by default now.

File: pkgs/development/compilers/openjdk/openjfx/17.nix [hunk 1]
```diff
@@ -14,6 +14,16 @@ let
...
 
+  icu-data = fetchurl {
+    url = "https://github.com/unicode-org/icu/releases/download/release-71-1/icu4c-71_1-data-bin-l.zip";
+    hash = "sha256-pVWIy0BkICsthA5mxhR9SJQHleMNnaEcGl/AaLi5qZM=";
+  };
+
+  fakeRepository = runCommand "icu-data-repository" {} ''
+    mkdir -p $out/download/release-71-1
+    cp ${icu-data} $out/download/release-71-1/icu4c-71_1-data-bin-l.zip
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/17.nix:25]: We should assert that the ico version matches

Comment (chvp) [reply pkgs/development/compilers/openjdk/openjfx/17.nix]: I used the version from the icu package to build the fake repository. This way the versions have to match.

**Comment by vdot0x23** [comment]
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux

File: pkgs/development/compilers/openjdk/openjfx/17.nix [hunk 2]
```diff
@@ -14,6 +14,18 @@ let
     java = openjdk17_headless;
   });
 
+  dashed-icu-version = lib.concatStringsSep "-" (lib.splitString "." (lib.getVersion icu71));
+  underscored-icu-version = lib.concatStringsSep "_" (lib.splitString "." (lib.getVersion icu71));
+  icu-data = fetchurl {
+      url = "https://github.com/unicode-org/icu/releases/download/release-${dashed-icu-version}/icu4c-${underscored-icu-version}-data-bin-l.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/openjfx/17.nix:20]:
```suggestion
    url = "https://github.com/unicode-org/icu/releases/download/release-${dashed-icu-version}/icu4c-${underscored-icu-version}-data-bin-l.zip";
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/compilers/openjdk/openjfx/17.nix (hunk 2 above)
Comment (chvp) [reply pkgs/development/compilers/openjdk/openjfx/17.nix]: Applied, thanks.

**Pull request merged**